crates/ty_python_semantic/src/types/class.rs

```diff
@@ -1592,7 +1592,9 @@ impl<'db> ClassLiteral<'db> {
 
         let field_policy = CodeGeneratorKind::from_class(db, self)?;
 
-        let signature_from_fields = |mut parameters: Vec<_>| {
+        let signature_from_fields = |mut parameters: Vec<_>,
+                                     return_ty: Option<Type<'db>>,
+                                     for_replace: bool| {
```

Comment:
didn't feel like the cleanest way to toggle some of this behavior, open to suggestions...